compiler/next/lib/queries/ErrorMessage.cpp

```diff
@@ -78,13 +78,13 @@ void ErrorMessage::addDetail(ErrorMessage err) {
 
 void ErrorMessage::swap(ErrorMessage& other) {
   std::swap(level_, other.level_);
-  this->location_.swap(other.location_);
-  this->message_.swap(other.message_);
-  this->details_.swap(other.details_);
+  location_.swap(other.location_);
+  message_.swap(other.message_);
+  details_.swap(other.details_);
 }
 
-void ErrorMessage::markUniqueStrings(Context* context) const {
-  this->location_.markUniqueStrings(context);
+void ErrorMessage::mark(Context* context) const {
+  this->location_.mark(context);
```

Comment:
not sure stance on `this->`, but with the above change in swap, consider dropping `this->` as well (that is my preference)